instrumentation/src/test/java/io/opentelemetry/android/OpenTelemetryRumBuilderTest.java

```diff
@@ -158,6 +168,69 @@ void setSpanExporterCustomizer() {
                 .untilAsserted(() -> verify(exporter).export(anyCollection()));
     }
 
+    @Test
+    void diskBufferingEnabled() {
+        PreferencesService preferences = mock();
+        CacheStorageService cacheStorage = mock();
+        doReturn(60 * 1024 * 1024L).when(cacheStorage).ensureCacheSpaceAvailable(anyLong());
+        setUpServiceManager(preferences, cacheStorage);
+        OtelRumConfig config = buildConfig();
+        config.setDiskBufferingConfiguration(
+                DiskBufferingConfiguration.builder().setEnabled(true).build());
+
+        OpenTelemetryRum.builder(application, config)
+                .addSpanExporterCustomizer(
+                        spanExporter -> {
+                            assertThat(spanExporter).isInstanceOf(SpanDiskExporter.class);
```

Comment:
Prefer not to have asserts nested in callbacks that might not have ever been called. This can lead to false sense of confidence. 😁  I think it's better to save the param in an `AtomicReference` in the callback and then do the assert after `build()` is called.

Reply:
Good point, it's updated now.